core/lib/dal/src/models/storage_sync.rs

```diff
@@ -154,12 +166,8 @@ impl ProtoFmt for Payload {
 
     fn read(message: &Self::Proto) -> anyhow::Result<Self> {
         let mut transactions = Vec::with_capacity(message.transactions.len());
-        for (i, tx) in message.transactions.iter().enumerate() {
-            transactions.push(
-                required(&tx.json)
-                    .and_then(|json_str| Ok(serde_json::from_str(json_str)?))
-                    .with_context(|| format!("transaction[{i}]"))?,
-            );
+        for tx in message.transactions.iter() {
+            transactions.push(tx.read()?)
```

Comment:
add error context, as it was previously

Reply:
Fixed, changes here: https://github.com/matter-labs/zksync-era/pull/1047/commits/d2bcc7edcfa6777e3353b14928a40e60a31a0f2c